src/pixie/images.nim

```diff
@@ -70,8 +70,11 @@ proc `[]=`*(image: Image, x, y: int, rgba: ColorRGBA) {.inline.} =
 
 proc fill*(image: Image, rgba: ColorRgba) =
   ## Fills the image with a solid color.
-  for c in image.data.mitems:
-    c = rgba
+  if rgba == rgba(0, 0, 0, 0):
+    zeroMem(image.data[0].addr, image.data.len * 4)
```

Comment:
special case is much faster, added benchmark:
fillOriginal: 0.4440748s
fill: 0.285684s
fillOriginal rgba(0, 0, 0, 0): 0.4603884s
fill rgba(0, 0, 0, 0): 0.1067025s